assets/js/blocks/price-filter/block.js

```diff
@@ -72,8 +72,10 @@ const PriceFilterBlock = ( { attributes, isPreview = false } ) => {
 
 	// Track price STATE changes - if state changes, update the query.
 	useEffect( () => {
-		debouncedUpdateQuery();
-	}, [ minPrice, maxPrice ] );
+		if ( ! attributes.showFilterButton ) {
+			debouncedUpdateQuery();
+		}
+	}, [ minPrice, maxPrice, attributes ] );
```

Comment:
Instead of setting `attributes` as a dependency you can directly set `attributes.showFilterButton`:
```suggestion
	}, [ minPrice, maxPrice, attributes.showFilterButton ] );
```

Reply:
agree with @Aljullu 's suggestion here, and that's also preferable because it reduces the number of times this block will be executed (attributes is a non-primitive so that will more frequently fail the reference equality comparison)